### PR TITLE
Csv import export

### PR DESCRIPTION
**Pull request by marthaklambert**

### CODE REVIEW

File: client/src/pages/SurveyDetail/ExportModal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  return stringToFilename;
+};
+
+const downloadSurvey = async (id, anonymous, title) => {
+  try {
+    axios.get(`/download/${id}/${anonymous}`).then((response) => {
+      FileDownload(response.data, `${createValidFilename(title)}.csv`);
+    });
+  } catch (error) {
+    // TODO put error message here
```
Comment (AntonioGargaro) [inline client/src/pages/SurveyDetail/ExportModal/index.jsx]: snackbar to go here!

File: client/src/pages/SurveyDetail/ExportModal/index.jsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      }}
+    >
+      <Box
+        style={{
+          top: `50%`,
+          left: `50%`,
+          transform: `translate(-50%, -50%)`,
+          height: `40vh`,
+          width: `50vw`,
+        }}
```
Comment (AntonioGargaro) [inline client/src/pages/SurveyDetail/ExportModal/index.jsx]: A lot of inline styling here, can it not be added to a class?

File: server/handlers/protected/uploadCSV.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      return res
+        .status(401)
+        .json({ message: 'No authorised user exists with that username.' });
+    }
+
+    return res.status(403).json({ message: "The JWT token isn't valid." });
+  })(req, res, next);
+};
+
+module.exports = uploadCSV;
```
Comment (AntonioGargaro) [inline server/handlers/protected/uploadCSV.js]: Maybe make an issue for future work to describe what this will actually do?

File: client/src/pages/SurveyDetail/ExportModal/index.jsx (hunk 1 above)
Comment (thomas-galligan) [inline client/src/pages/SurveyDetail/ExportModal/index.jsx]: can you do this?

File: server/dummyData/dummySurveys.js [hunk 4]
```diff
@@ -245,7 +245,7 @@ module.exports = [
     datePublished: 1574259113,
     dateToClose: 1576851113,
     dateClosed: 1576851113,
-    anonymous: true,
+    anonymous: false,
```
Comment (thomas-galligan) [inline server/dummyData/dummySurveys.js]: why is this necessary?

File: server/handlers/protected/downloadCSV.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            if (currentQuestion.commentEnabled) {
+              responseObj[`Comment on ${currentQuestion.title}`] =
+                answer.comment;
+            }
+          });
+          return responseObj;
+        });
+        const json2csvParser = new Parser({ fields });
+        const csv = json2csvParser.parse(responsesWithQuestionTitles);
+        const file = `${__dirname}/../../temp/sample2return.csv`;
```
Comment (thomas-galligan) [inline server/handlers/protected/downloadCSV.js]: you should probs do path.join here

File: server/dummyData/dummySurveys.js (hunk 4 above)
Comment (marthaklambert) [reply server/dummyData/dummySurveys.js]: the survey says it is anonymous however responses are recorded with employee ids

File: client/src/pages/SurveyDetail/ExportModal/index.jsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+const createValidFilename = (string) => {
+  // currently a rough implementation that just replaces everything with an underscore, this could be refined
+  const stringToFilename = string.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+  return stringToFilename;
+};
+
+const downloadSurvey = async (id, anonymous, title) => {
+  try {
+    axios.get(`/download/${id}/${anonymous}`).then((response) => {
```
Comment (Joaoviana) [inline client/src/pages/SurveyDetail/ExportModal/index.jsx]: get response from `await` then do filedownload

File: client/src/pages/SurveyDetail/ExportModal/index.jsx (hunk 1 above)
Comment (marthaklambert) [reply client/src/pages/SurveyDetail/ExportModal/index.jsx]: being resolved by Toni

Comment (marthaklambert) [reply client/src/pages/SurveyDetail/ExportModal/index.jsx]: being resolved by you

File: client/src/utils/checkTokenIsAuth.js [hunk 7]
```diff
@@ -9,6 +9,7 @@ const isTokenAuthed = async (dispatch, currentAuth) => {
       headers: { Authorization: `JWT ${token}` },
     });
     if (res && res.status === 200 && currentAuth === false) {
+      console.log('authorising');
```
Comment (lyndseysc) [inline client/src/utils/checkTokenIsAuth.js]: do we need this log

File: server/handlers/protected/downloadCSV.js [hunk 8]
```diff
@@ -2,46 +2,78 @@ const passport = require('passport');
...
+          }
+
+          const responsesWithQuestionTitles = responses.map((response) => {
+            const responseObj = {};
+            if (!anonymous) {
+              responseObj.Name = employees.find(
+                (employee) => response.employeeId === employee._id,
+              );
+            }
+            console.log(responseObj, 'RESP OBJ');
```
Comment (lyndseysc) [inline server/handlers/protected/downloadCSV.js]: still many logs
